kernels/volk/volk_32f_cos_32f.h

```diff
@@ -673,4 +673,41 @@ volk_32f_cos_32f_generic(float* bVector, const float* aVector, unsigned int num_
 
 #endif /* LV_HAVE_GENERIC */
 
+
+#ifdef LV_HAVE_NEON
+#include <arm_neon.h>
+#include <volk/volk_neon_intrinsics.h>
+
+static inline void
+volk_32f_cos_32f_neon(float* bVector, const float* aVector,
+                      unsigned int num_points)
+{
+    unsigned int number = 0;
+    unsigned int quarter_points = num_points / 4;
+    float* bVectorPtr = bVector;
+    const float* aVectorPtr = aVector;
+    
+    float32x4_t b_vec;
+    float32x4_t a_vec;
+    
+    for(number = 0; number < quarter_points; number++) {
+        a_vec = vld1q_f32(aVectorPtr);
+        // Prefetch next one, speeds things up
+        __VOLK_PREFETCH(aVectorPtr+4);
```

Comment:
On the final iteration of this loop, `aVectorPtr+4` will point to the memory just beyond the end of the end of the `aVectorPtr`. There's no guarantee that this memory will have been allocated contiguously, and hence this prefetch could, maybe possibly, generate a runtime memory access error. We need a better way to do this prefetch! How much of a hit does this code take when the prefetch is disabled?

Reply:
Good find. Probably not a big hit.

A quick `grep` reveals this bug is present in other kernels (also kernels I didn't touch).

I can't think of a way of pre-fetching without using a test. Probably best to remove them.